src/reducedStream.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  Subject,
+  SubscriptionLike,
+  Operator,
+  Subscriber,
+  TeardownLogic,
+} from 'rxjs';
+import { ActionStream } from './types/helpers';
+import { RegisteredReducer, combineReducers } from './reducer';
+import { defaultErrorSubject } from './internal/defaultErrorSubject';
+import { markName } from './internal/markers';
+import { tag } from 'rxjs-spy/operators';
+
+export class ReducedStateStream<State> extends Observable<State>
+  implements SubscriptionLike {
```

Comment:
Looking at this again this is strange stuff. The Observable class is only used as an interface, wrapping the actual stream as a private subject member. The interface seems only partially implemented. It's difficult to really understand what this class tries to achieve. Also things are not really symmetric anymore, like subscribe and unsubscribe, subscribe does not start the actual subscription, but unsubscribe actually stops it.

Reply:
I see your point, but I had to be "creative" in order to achieve the intended behaviour, as I was unable to find another way. Do you have suggestions for how this could be improved?

Reply:
I don't really understand the problem we are trying to solve here. Could you describe it?

Reply:
The minimum would IMO be though to not use `unsubscribe` but rather `stopReducing` to keep it symmetric, otherwise this is doomed to cause confusion, I think.

Reply:
Sure, we are trying to solve a set of problems related to creating persistent state streams in an application:

1. setting up & consuming state streams should be simple
    a. their values can be read synchronously (using `.state` like a behaviour subject)
    b. their values can be streamed (like any observable)
2. the streams are persistent, which means that we wish to subscribe to them and keep them "alive" through the application's lifecycle
3. they can be easily tested
4. they can be hot-reloaded during development

Let's look at each point separately. 

1. With the previous solution, I think we agreed that consuming & setting up streams was no longer simple, since the stream [had to be wrapped in get and setup methods](https://github.com/ardoq/ardoq-front/blob/aacb99716ce545fcd015ec96bfc6965bf191ca13/src/js/ardoq/branchId%24.ts#L16-L35). This PR fixes this by introducing a stream-like constructor "PersistentReducedStateStream". 
    - Consuming the stream is now straight-forward and simple. You can access the stream's value either by streaming it or by accessing `.state` synchronously. This interface is very similar to a behaviour subject (which inspired the solution).
    - Setting up the stream still requires [an additional step of calling startReducing](https://github.com/ardoq/ardoq-front/pull/7135/files#diff-ff0d9d26e235633b69491de7bc7b6cf5R31-R33), but this is unavoidable if we wish to maintain control of when persistent reduced streams are started.
2. Previously, we made streams persistent by `subscribing` and never removing this subscription. Now we do this by calling `startReducing` which emulates this setup step.
3. With this solution, the [state stream definition](https://github.com/ardoq/ardoq-front/pull/7135/files#diff-5dbdaf1d6e2303c79deffb56019a573d) file should ideally have no dependencies except pure actions & types. The action stream is injected using `startReducing` which makes it trivial to inject a mocked action stream. Unfortunately this is not the case for stream reducers yet, but this is difficult to solve while maintaining strict typing, so they might have to be mocked otherwise.
4. Finally, the proposed interface makes it trivial to solve hot-reloading, since we can read the state synchronously from the stream when it's hot-reloaded and [restart reducing of the stream with the latest value](https://github.com/ardoq/ardoq-front/blob/3b6e60d7b2c5d30f22057da923f668c8faf3a6ca/src/js/ardoq/startStateStreams.ts#L62-L90). 

Perhaps each point needs a bit more explanation & discussion, I'm happy to do that in a call :) From my point of view, it's very important to keep consumption of state streams simple and I think this PR achieves that. The internals of the stream could possibly be improved, but I think the idea of representing our "state streams" as custom observables similar to a behaviour subject makes sense. 

In conclusion, from the "ardoq-front" side, I think this implementation is the best we have, because:
- consuming streams is simple - you consume them like any other stream with the added benefit of having a `.state` accessor. 
- setting up streams  has the small caveat of calling `startReducing`, but I think we can accept this downside because it allows us to implement hot reloading & it makes testing simple

If you have any suggestions for how to improve the interface or think we are on the complete wrong path, that feedback is of course very welcome!

Reply:
Could we not achieve the same thing by simply extending BehaviorSubject, adding our custom "teardown" logic and possibly adding a hook to register the stream for "startReducing" (we could store them in a registry, that way we wouldn't have to import all of them in `startStateStreams.ts`, just that registry module)?

This is just by reading your explanation -- I haven't looked at the implementation in detail, although I agree with Krebs that the class is a bit weird right now.

Reply:
@tlaundal 's initial PR extended BehaviorSubject, but we [landed upon extending Observable](https://github.com/ardoq/rxbeach/pull/142#discussion_r411507053) since it made more sense to expose state using `.state` and we didn't see any pros for extending behavior subject. I think one of main arguments was that the interface we've created is no longer a behavior subject, it's rather an interface to an internal behavior subject 🤓 😄 

Regarding a stream registry, would be nice with a concrete example that shows what you mean. I'm guessing that would require creating a static shared stream registry module among all the persistent reduced streams? Would we still be able to support hot reloading and controlling when reducing is started? Sounds like the approach could be a nice improvement if it removes the need for importing all of the streams in `startStateStreams` :)

Reply:
Had a live discussion with @chriskr. I will look into the `unsubscribe` method and possibly split it into `stopReducing` and `unsubscribe` to make the interface more consistent & "symmetrical".

Reply:
Yes, "static" registry. How things would be registered is up for discussion, but I think it would "fix" startStateStreams. Not sure how it impacts hot reloading because I'm not that familiar with the hot reloading mechanism.